tests/test_appsettings.py

```diff
@@ -426,165 +521,141 @@ def test_nested_dict_setting_required_both_inner_and_outer_setting(self):
-
-    @mock.patch.dict(os.environ, {"BOOL_LOWER": "false", "BOOL_UPPER": "FALSE", "BOOL_NUM": "0", "BOOL_WORD": "no"})
-    def test_string_boolean_setting_from_environ_false_value(self):
-        bool_lower = appsettings.BooleanSetting(name="bool_lower")
-        bool_lower.check()
-        assert bool_lower.value is False
-
-        bool_upper = appsettings.BooleanSetting(name="bool_upper")
-        bool_upper.check()
-        assert bool_upper.value is False
-
-        bool_num = appsettings.BooleanSetting(name="bool_num")
-        bool_num.check()
-        assert bool_num.value is False
-
-        bool_word = appsettings.BooleanSetting(name="bool_word")
-        bool_word.check()
-        assert bool_word.value is False
-
-    @mock.patch.dict(os.environ, {"BOOL_SETTING": "invalid"})
-    def test_string_boolean_setting_from_environ_invalid_value(self):
-        bool_setting = appsettings.BooleanSetting(name="bool_setting")
-        with pytest.raises(ValueError, match="Invalid boolean setting BOOL_SETTING"):
-            bool_setting.check()
-
-    @mock.patch.dict(os.environ, {"SETTING": "123"})
-    def test_integer_setting_from_environ_value(self):
-        setting = appsettings.IntegerSetting(name="setting")
+    @mock.patch.dict(os.environ, {"SETTING": '{"A": "A", "B": "B"}'})
+    def test_nested_dict_setting_from_environ_value(self):
+        setting = appsettings.NestedDictSetting(
+            settings=dict(a=appsettings.Setting(), b=appsettings.Setting(),), name="setting"
+        )
         setting.check()
-        assert setting.value == 123
-        assert type(setting.value) is int
+        assert setting.value == {"a": "A", "b": "B"}
 
-    @mock.patch.dict(os.environ, {"SETTING": "123.456"})
-    def test_float_setting_from_environ_value(self):
-        setting = appsettings.FloatSetting(name="setting")
+    @mock.patch.dict(os.environ, {"SETTING": '{"A": "A", "B": "B"}', "A": "Fake A", "B": "Fake B"})
+    def test_parent_setting_precedence_over_environ_value(self):
+        setting = appsettings.NestedDictSetting(
+            settings=dict(a=appsettings.StringSetting(), b=appsettings.StringSetting(),), name="setting"
+        )
         setting.check()
-        assert setting.value == 123.456
-        assert type(setting.value) is float
+        assert setting.value["a"] == "A"
+        assert setting.value["b"] == "B"
 
-    @mock.patch.dict(os.environ, {"SETTING": "[1, 2, 3]"})
-    def test_iterable_setting_from_environ_json_value(self):
-        setting = appsettings.IterableSetting(name="setting")
-        setting.check()
-        assert setting.value == [1, 2, 3]
 
-    @mock.patch.dict(os.environ, {"SETTING": "1:2:3"})
-    def test_iterable_setting_from_environ_delimiter_value(self):
-        setting = appsettings.IterableSetting(name="setting")
-        setting.check()
-        assert setting.value == ["1", "2", "3"]
+class NestedSettingTestCase(SimpleTestCase):
+    """NestedSetting tests."""
```

Comment:
This should be probably part of the `NestedDict` tests.

Reply:
Moved.